packages/react-components/react-button/src/components/Button/useButton.ts

```diff
@@ -23,8 +23,21 @@ export const useButton_unstable = (
     iconPosition = 'before',
     shape = 'rounded',
     size = contextSize ?? 'medium',
+    actionState = 'none',
+    onClick,
   } = props;
   const iconShorthand = slot.optional(icon, { elementType: 'span' });
+
+  const _props = { ...props };
+
+  _props.onClick = useEventCallback(
+    (ev: React.MouseEvent<HTMLButtonElement, MouseEvent> & React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
```

Comment:
This should be OR, because an event cannot be both Button and Anchor. :)
```suggestion
    (ev: React.MouseEvent<HTMLButtonElement, MouseEvent> | React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
```